packages/ibm-products/src/components/ConditionBuilder/ConditionBuilderActions/ConditionBuilderActions.js

```diff
@@ -0,0 +1,148 @@
+import React, { useContext } from 'react';
+import PropTypes from 'prop-types';
+import { Close } from '@carbon/react/icons';
+import { Section, Heading } from '@carbon/react';
+import { ConditionBuilderItem } from '../ConditionBuilderItem/ConditionBuilderItem';
+import {
+  blockClass,
+  translateWithId,
+} from '../ConditionBuilderContext/DataConfigs';
+import { ItemOption } from '../ConditionBuilderItem/ConditionBuilderItemOption/ItemOption';
+import { ConditionBuilderContext } from '../ConditionBuilderContext/ConditionBuilderProvider';
+import ConditionBuilderAdd from '../ConditionBuilderAdd/ConditionBuilderAdd';
+import uuidv4 from '../../../global/js/utils/uuidv4';
+import { ConditionBuilderButton } from '../ConditionBuilderButton/ConditionBuilderButton';
+import { checkDuplicateAction, focusThisField } from '../utils/util';
+
+const ConditionBuilderActions = ({ actions, className }) => {
+  const { actionState, setActionState } = useContext(ConditionBuilderContext);
+
+  const addActionHandler = () => {
+    const action = {
+      id: uuidv4(),
+      label: undefined,
+      popoverToOpen: 'actionField',
+    };
+    setActionState([...actionState, action]);
+  };
+
+  const onchangeHandler = (evt, selectedId, actionIndex, currentActionId) => {
```

Comment:
Hey,
Did some refractor as we add same action again rather than showing validation.
Also slice what we have used on the other places of this component.